### PR TITLE
Remove qchem deprecated functionality for fermionic observables

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
The deprecated qchem functions and features that were using tuples to represent fermionic observables are removed. The relevant tests are updated.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-43900]

**Comment by lillian542** [comment]
[sc-43904]

**Comment by lillian542** [comment]
[sc-43905]

File: pennylane/fermi/conversion.py [hunk 1]
```diff
@@ -13,15 +13,10 @@
...
 from functools import singledispatch
 from typing import Union
-import numpy as np
-
-import pennylane as qml
-from pennylane.operation import Operator, Tensor
+from pennylane.operation import Operator
 from pennylane.pauli import PauliWord, PauliSentence
-from pennylane.pauli.utils import _get_pauli_map, _pauli_mult
 from .fermionic import FermiWord, FermiSentence
```
Comment (lillian542) [inline pennylane/fermi/conversion.py:22]: now that we don't have the dispatch with the different call signature, I think we can update this from `fermi_operator: (Union[FermiWord, FermiSentence]), **kwargs` to `fermi_operator: (Union[FermiWord, FermiSentence]), ps: bool = False, wire_map: dict = None`

File: pennylane/qchem/hamiltonian.py [hunk 2]
```diff
@@ -180,27 +176,19 @@ def _fermionic_hamiltonian(*args):
```
Comment (lillian542) [inline pennylane/qchem/hamiltonian.py:176]: The Returns section above should be updated to only `FermiSentence`

File: pennylane/qchem/observable_hf.py [hunk 3]
```diff
@@ -14,26 +14,22 @@
...
 
     Args:
         constant (array[float]): the contribution of the core orbitals and nuclei
         one (array[float]): the one-particle molecular orbital integrals
         two (array[float]): the two-particle molecular orbital integrals
         cutoff (float): cutoff value for discarding the negligible integrals
-        fs (bool): if True, a fermi sentence will be returned
 
     Returns:
         Union[FermiSentence, tuple(array[float], list[list[int]])]: fermionic Hamiltonian
```
Comment (lillian542) [inline pennylane/qchem/observable_hf.py:35]:
```suggestion
        FermiSentence: fermionic Hamiltonian
```

File: pennylane/qchem/observable_hf.py [hunk 4]
```diff
@@ -166,91 +131,22 @@ def qubit_observable(o_ferm, cutoff=1.0e-12):
     >>> print(qubit_observable(f))
```
Comment (lillian542) [inline pennylane/qchem/observable_hf.py:131]: lines 128-130 still need to be updated to use `FermiWord` and `FermiSentence`

**Comment by lillian542** [comment]
Also needs a changelog entry and we need to update the deprecations page :)

File: doc/development/deprecations.rst [hunk 5]
```diff
@@ -69,6 +69,15 @@ Pending deprecations
...
+* The ``QuantumScript.set_parameters`` method and the ``QuantumScript.data`` setter has
+  been deprecated. Please use ``QuantumScript.bind_new_parameters`` instead.
+
+  - Deprecated in v0.32
+  - Will be removed in v0.33
+
+Completed deprecation cycles
+----------------------------
+
 * ``qml.qchem.jordan_wigner`` is deprecated, and usage will now raise a warning.
```
Comment (lillian542) [inline doc/development/deprecations.rst:82]:
```suggestion
* ``qml.qchem.jordan_wigner`` had been removed.
```

File: doc/development/deprecations.rst [hunk 6]
```diff
@@ -69,6 +69,15 @@ Pending deprecations
...
   Use ``qml.jordan_wigner`` instead. List input to define the fermionic operator
   is also deprecated; the fermionic operators ``qml.FermiA``, ``qml.FermiC``,
```
Comment (lillian542) [inline doc/development/deprecations.rst:84]:
```suggestion
 is no longer accepted; the fermionic operators ``qml.FermiA``, ``qml.FermiC``,
```

**Review by lillian542 — APPROVED** [review]
A couple of small suggestions for the changelog, but looks good to me!

**Review by Jaybsoni — APPROVED** [review]
Looks good to me 💯

**Pull request merged**